src/components/bar-chart/bar-chart.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, ElementRef, Input, OnChanges, OnDestroy, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { LegendItemStyle, BarChartD3 } from '../../d3/bar-chart.d3';
+import { RenderOptions } from '../../d3/xy-chart.d3';
+import { GUIDE_DOGE, t } from '../../i18n';
+import { formatX, formatY } from '../../utils/formatters';
+import { A11yPlaceholderDirective } from '../../directives/a11y-placeholder/a11y-placeholder.directive';
+import { BarChartMeta } from '../../datasets/metas/categorical.meta';
+import { CategoricalDatum } from '../../datasets/queries/categorical.query';
+import { CategoricalPoint } from '../../datasets/metas/types';
+import { A11yHostComponent } from '../a11y-host/a11y-host.component';
+
+export type BarChartDatum = CategoricalDatum<LegendItemStyle>;
+
+@Component({
+  selector: 'app-bar-chart',
+  templateUrl: './bar-chart.component.html',
+  styleUrls: ['./bar-chart.component.scss'],
+})
+export class BarChartComponent extends A11yHostComponent implements
+  RenderOptions<CategoricalPoint, BarChartDatum>, OnChanges, OnInit, OnDestroy {
+  @ViewChild(A11yPlaceholderDirective, { static: true }) a11yPlaceholder: A11yPlaceholderDirective;
+
+  @Input() meta: BarChartMeta;
+  @Input() height = 500;
+  @Input() width = 800;
+
+  formatX = formatX;
+  formatY = formatY;
+
+  data$ = new BehaviorSubject<CategoricalDatum<LegendItemStyle>[]>([]);
+  barChartD3: BarChartD3;
+  private destroy$ = new Subject();
+
+  constructor(
+    public elementRef: ElementRef<HTMLElement>,
+  ) {
+    super(elementRef);
+    this.barChartD3 = new BarChartD3(this);
+  }
+
+  get data() {
+    return this.data$.value;
+  }
+
+  get VISUALIZATION() {
+    return t(GUIDE_DOGE.VISUALIZATION);
```

Comment:
nit: change "t" to something for readable

Reply:
`t()` is the core function in i18n of guide-doge, and I would suggest not to change it. The audification, geo-map, and dashboard controls heavily use these i18n functions, and changing them might cause some errors which out of my control and hard for me to resolve quickly.